--- gargoyle-gitrepo/target/linux/ramips/files/arch/mips/ralink/rt305x/setup.c	2013-05-17 14:48:48.620619101 +0800
+++ shenrui/target/linux/ramips/files/arch/mips/ralink/rt305x/setup.c	2013-05-23 22:08:19.943940993 +0800
@@ -25,6 +25,8 @@
 #include <asm/mach-ralink/rt305x_regs.h>
 #include "common.h"
 
+static unsigned int ramips_cpu_freq;
+
 static void rt305x_restart(char *command)
 {
 	rt305x_sysc_wr(RT305X_RESET_SYSTEM, SYSC_REG_RESET_CTRL);
@@ -58,6 +60,7 @@
 	if (IS_ERR(clk))
 		panic("unable to get CPU clock, err=%ld", PTR_ERR(clk));
 
+	ramips_cpu_freq = clk_get_rate(clk);
 	printk(KERN_INFO "%s running at %lu.%02lu MHz\n", ramips_sys_type,
 		clk_get_rate(clk) / 1000000,
 		(clk_get_rate(clk) % 1000000) * 100 / 1000000);
@@ -86,3 +89,8 @@
 
 	mips_hpt_frequency = clk_get_rate(clk) / 2;
 }
+
+unsigned int ramips_get_cpu_freq(void)
+{
+	return ramips_cpu_freq;
+}
